sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -7344,6 +7344,36 @@ public void testUsingSubqueryWithExtractionFns() throws Exception
     );
   }
 
+  @Test
+  public void testUsingSubqueryWithLimit() throws Exception
+  {
+    testQuery(
+        "SELECT COUNT(*) AS cnt FROM ( SELECT * FROM druid.foo LIMIT 10 ) tmpA",
```

Comment:
Whoa, this looks wrong. The query doesn't have a native query and it returns an empty result set.

I think that probably, this query should return an error, because it represents a kind of query structure Druid can't do right now (aggregation on top of a non-aggregating subquery).

Reply:
In fact, there is no exception thrown here. Normally, the `DruidOuterQueryRel#toDruidQuery` called in the `DruidOuterQueryRel#runQuery` method will return a non-empty DruidQuery instance, after which the program will go to the first branch of if-else, call `QueryMaker#runQuery` and perform a structural check. However, as you said before, when GroupByQuery is empty, `DruidOuterQueryRel#toDruidQuery` will return null directly, so that the program will go to the second branch of if-else, no check will be performed, it will be directly Return to Sequences.empty(). Therefore, no error will be thrown. :thinking:

Reply:
FYI, the code logic I am looking at is https://github.com/apache/incubator-druid/pull/8775/files#diff-fbc88cca6610397d841e9b32a9a86f13R89-R101

Reply:
Since Druid can't support this type of query right now, I think this could throw a `CannotBuildQueryException` if the subquery can't be expressed as a groupBy here: https://github.com/apache/incubator-druid/pull/8775/files#diff-fbc88cca6610397d841e9b32a9a86f13R133

Reply:
@jon-wei Thanks a lot. It's done. PTAL.